emulator/src/extensions/tools/ext_tools.c

```diff
@@ -48,21 +63,129 @@ static const char* getScreenshotName(char* buffer, size_t maxlen);
 
 emuExtension_t toolsEmuExtension = {
     .name            = "tools",
-    .fnInitCb        = NULL,
-    .fnPreFrameCb    = NULL,
-    .fnPostFrameCb   = NULL,
+    .fnInitCb        = toolsInit,
+    .fnPreFrameCb    = toolsPreFrame,
+    .fnPostFrameCb   = toolsPostFrame,
     .fnKeyCb         = toolsKeyCb,
     .fnMouseMoveCb   = NULL,
     .fnMouseButtonCb = NULL,
-    .fnRenderCb      = NULL,
+    .fnRenderCb      = toolsRenderCb,
 };
 
+static bool useFakeTime       = false;
+static uint64_t fakeTime      = 0;
+static uint64_t fakeFrameTime = 0;
+
+static bool recordScreen           = false;
+static char recordingFilename[256] = {0};
+
+static bool pauseNextFrame = false;
+
+static bool showFps  = false;
```

Comment:
`make format` please!

Reply:
It is formatted! I think `clang-format` now breaks alignment at empty lines and such that it didn't do before

Reply:
Ah, it always reset alignment at empty lines. There isn't an empty line here, but `frameStartTime` doesn't have an initial value, and that's resetting alignment too. TIL.